src/proc_cpuview.c

```diff
@@ -232,6 +234,31 @@ static struct cg_proc_stat *prune_proc_stat_list(struct cg_proc_stat *node)
 		if (!cgroup_supports("cpu", node->cg, "cpu.shares")) {
 			struct cg_proc_stat *cur = node;
 
+			/*
+			 * We need to ensure that no one referenced this node,
+			 * because we are going to remove it from the list and free memory.
```

Comment:
In the long run one would probably want a refcount pattern for this, i.e., have users do:

lock()
refcount_inc_not_zero()
unlock()

and have writers/deleters do

lock()
refcount_dec_and_test()
unlock()

But that would require a bigger redesign of this thing. Really what you'd want is something like a userspace rcu library which exists already but anyway, looks good enough as a fix.

Reply:
>But that would require a bigger redesign of this thing. Really what you'd want is something like a userspace rcu library which exists already but anyway, looks good enough as a fix.

Agreed totally. We can take a look on that in near future :-)